plans/objchange/plan_valid.go

```diff
@@ -39,13 +39,10 @@ func AssertPlanValid(schema *configschema.Block, priorState, config, plannedStat
 func assertPlanValid(schema *configschema.Block, priorState, config, plannedState cty.Value, path cty.Path) []error {
 	var errs []error
 	if plannedState.IsNull() && !config.IsNull() {
-		errs = append(errs, path.NewErrorf("planned for absense but config wants existence"))
-		return errs
-	}
-	if config.IsNull() && !plannedState.IsNull() {
-		errs = append(errs, path.NewErrorf("planned for existence but config wants absense"))
+		errs = append(errs, path.NewErrorf("planned for absence but config wants existence"))
 		return errs
 	}
+
```

Comment:
Rather than weakening this check in general, could we not make the `timeouts` attribute be null if it's not set in config?

Ideally a block should appear in the plan only if it was present in the config; "computed blocks" aren't a thing, and so in our new-SDK world if a provider wants to generate "default" objects then it should do so in a computed attribute of object type, rather than in a block.

Historical SDK conventions prevent us from enforcing that right now, but since `timeouts` is being added by the shims rather than by legacy SDK code I would expect it to respect our new conventions.

Reply:
Yeah, I was just thinking about ways future providers could get around this, and if we want to enforce that a null block must remain null, then I think the original check should stay.

However, I think inserting the timeouts block into the legacy config is going to add more noise, since every plan would then plan with "unknown" for create, read, update, delete, and default.